shacl12-node-expr/index.html

```diff
@@ -2647,6 +2647,142 @@ <h3>sh:nodeByExpression</h3>
+					of a president is 18 unless the country is USA, where it is 35.
+				</p>
+				<aside class="example" title="Example of Dynamic SHACL using a node expression at sh:minInclusive">
+					<div class="shapes-graph">
+						<div class="turtle">
+ex:PresidentShape
+    a sh:NodeShape ;
+	sh:targetClass ex:President ;
+    sh:property ex:PresidentShape-age ;
+.
+ex:PresidentShape-age
+    a sh:PropertyShape ;
+    sh:path ex:age ;
+    <b>sh:minInclusive [
+        shnex:if [
+            sparql:eq (
+                [ shnex:path ex:country ]
+                ex:USA
+            )
+        ]
+        shnex:then 35 ;
+        shnex:else 18 ;
+    ]</b> .
+						</div>
+						<div class="jsonld">
+							<pre class="jsonld">
+								TODO
+							</pre>
+						</div>
+					</div>
+				</aside>
+			</section>
+			<section class="informative">
+				<h3>Example: Dynamic Enumerations</h3>
+				<p>
+					As a use case of <a>Dynamic SHACL</a>, assume the following <a>data graph</a>.
+				</p>
+				<aside class="example" title="Example data graph with addresses">
+					<div class="data-graph">
+						<div class="turtle">
+ex:ArizonaAddress1
+    a ex:Address ;
+    ex:street "123 John Muir Ave" ;
+    ex:country ex:USA ;
+    ex:state "AZ" ;
+.
+ex:QueenslandAddress1
+    a ex:Address ;
+    ex:street "123 Bob Katter Cl" ;
+    ex:country ex:Australia ;
+    ex:state "QLD" ;
+.
+						</div>
+						<div class="jsonld">
+							<pre class="jsonld">
+								TODO
+							</pre>
+						</div>
+					</div>
+				</aside>
+				<p>
+					We want to express that the valid values of <code>ex:state</code> depend on the value of <code>ex:country</code>
+					at the given focus node.
+					For example, the valid values for country <code>ex:USA</code> would be <code>( "AL" "AK" "AZ" ... )</code>
+					while values values for country <code>ex:Australia</code> would be <code>( "ACT" "NSW" "NT" "QLD" "SA" "TAS" "VIC" "WA" )</code>.
+					This fact can be represented as part of the data:
+				</p>
+				<aside class="example" title="The valid states can be attached to each country">
+					<div class="data-graph">
+						<div class="turtle">
+ex:Australia
+    a ex:Country ;
+    ex:stateCode "ACT", "NSW", "NT", "QLD", "SA", "TAS", "VIC", "WA" ;
+.
+ex:USA
+    a ex:Country ;
+    ex:stateCode "AL", "AK", "AZ", "AR" ; # ...
+.
+						</div>
+						<div class="jsonld">
+							<pre class="jsonld">
+								TODO
+							</pre>
+						</div>
+					</div>
+				</aside>
+				<p>
+					Using this extra information, we can now define a <code>sh:in</code> constraint using a <a>Path Expression</a>:
+				</p>
+				<aside class="example" title="Example of Dynamic SHACL using a node expression at sh:in">
+					<div class="shapes-graph">
+						<div class="turtle">
+ex:Address
+    a sh:ShapeClass ;
+    sh:property ex:Address-state ;
+.
+ex:Address-state
+    a sh:PropertyShape ;
+    sh:path ex:state ;
+    <b>sh:in [
+        shnex:path ( ex:country ex:stateCode )
+    ]</b> .
```

Comment:
```suggestion
ex:Address
    a           sh:ShapeClass ;
    sh:property ex:Address-state ;
.
ex:Address-state
    a           sh:PropertyShape ;
    sh:path     ex:state ;
    <b>sh:in [
        shnex:path ( ex:country ex:stateCode )
    ]</b> .
```

Reply:
Closing a change request without comment is *not* a good way to "resolve conversation". Please do not do so in future. If you disagree with a change request, either reply to it in place, or use the `...` menu at upper-right to `Reference in a new issue`.

(I find that GitHub's default `Reference in a new issue` boilerplate is sub-optimal, and so I edit it when I do this, but the default is functional if imperfect, so it's fine to just take it as they give it.)

Reply:
```
ex:PresidentShape
    a               sh:NodeShape ;
```
I don't think this is an improvement.

Reply:
maybe it looks better in the browser? 🤔